### PR TITLE
[acrn-configuration-tool] Add schema of scenario xml

### PR DESCRIPTION
**Pull request by yuchuyang**
Introduce the schemas with file extension "xsd" which is capable to validate the xml formatting against schema.
The xsd file provides the desired input data format, data type and description for elements.

Tracked-On: #5672
Signed-off-by: Yang,Yu-chu <yu-chu.yang@intel.com>

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -48,6 +48,9 @@ doxy: copy-to-sourcedir
 
 content: copy-to-sourcedir
 	$(Q)scripts/extract_content.py $(SOURCEDIR) misc
+	$(Q)mkdir -p $(SOURCEDIR)/misc/acrn-config/schema
+	$(Q)rsync -rt ../misc/acrn-config/schema/*.xsd $(SOURCEDIR)/misc/acrn-config/schema
+	$(Q)xsltproc -xinclude ./scripts/configdoc.xsl $(SOURCEDIR)/misc/acrn-config/schema/config.xsd > $(SOURCEDIR)/reference/configdoc.txt
```
Comment (NanlinXie) [inline doc/Makefile:53]: Please modify based on new folder org, acrn-config -> config_tools

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Comment by wenlingz** [comment]
@yuchuyang about doc part, please create new one PR for it. Thanks. Usually we need separate doc PR from code PR.

File: doc/reference/config-options.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  A base-10 value represented by the characters ``0`` to ``9``.  The
+  first character must not be a ``0``. Only positive values are
+  expected.
+
+String
+  A sequence of UTF-8 characters.  String-length limits or specific
+  string value restrictions are defined in the option description.
+
+
+.. comment This configdoc.txt is generated during the doc build process
```
[the remark this answers is not in the document]
Comment (dbkinder) [reply doc/reference/config-options.rst:38]: technically you're right, the "comment" is not needed, but it's more obvious that it's a comment to I tend to add it.

File: doc/scripts/genrest.py [hunk 3]
```diff
@@ -50,8 +50,8 @@ def expr_str(expr):
 
 INDEX_RST_HEADER = """.. _configuration:
 
-Configuration Symbol Reference
-##############################
+Kconfig Symbol Reference
```
[the remark this answers is not in the document]
Comment (dbkinder) [reply doc/scripts/genrest.py:53]: In a subsequent PR we'll be completely removing all the Kconfig option documents and this script.  Kconfig (and ``make menuconfig``) are not part of the new configuration process.

File: misc/config_tools/schema/types.xsd [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  <xs:union memberTypes="SosRamSize HexFormat" />
+</xs:simpleType>
+
+<xs:simpleType name="LogLevelType">
+  <xs:annotation>
+    <xs:documentation>An integer from 0 to 7 representing log message
+severity and intent:
+
+- 1 (LOG_FATAL) system is unusable
+- 2 (LOG_ACRN)
```
[the remark this answers is not in the document]
Comment (dbkinder) [reply misc/config_tools/schema/types.xsd:70]: no idea :)  All of these descriptions in the xsd files were derived from Kconfig or documentation descriptions I could find.  There will be a thorough review for technical accuracy, clarity, and formatting once we have the initial commit merged and we can do more surgical editing.

**Comment by dbkinder** [comment]
@gvancuts @yuchuyang @NanlinXie I force pushed changes to yuchuyan's branch to fix Geoffroy's spelling fixes (and a few more), as well as some formatting errors that were causing the doc build to fail.  The docs build without error locally, but the CI system does not have ``xsltproc`` installed so the doc build is still failing.  @wenlingz can you verify you added ``xsltproc`` to CI?

**Review by dbkinder — APPROVED** [review]
I forced pushed changes to your branch to address Geoffroy's comments and fix doc build errors.  We'll continue reviewing and ending the descriptions of the config options in subsequent PRs.

**Pull request merged**